src/renderer/screens/exchange/Buy/index.js

```diff
@@ -1,96 +1,102 @@
-    });
-  }, []);
+const Buy = (props: DProps) => {
+  const [selected, setSelected] = useState<Provider>(null);
+  const [isCoinify, setIsCoinify] = useState(false);
+  const { t } = useTranslation();
+  const history = useHistory();
 
-  const confirmAccount = useCallback((account: AccountLike, parentAccount: Account) => {
-    setState(oldState => ({
-      ...oldState,
-      account: account,
-      parentAccount: parentAccount,
-    }));
+  const toggle = useCallback((provider: Provider) => {
+    setSelected(prev => (prev === provider ? null : provider));
   }, []);
 
-  const selectAccount = useCallback(
-    (account, parentAccount) => {
-      dispatch(
-        openModal("MODAL_EXCHANGE_CRYPTO_DEVICE", {
-          account,
-          parentAccount,
-          onResult: confirmAccount,
-          flow: "buy",
-        }),
-      );
-    },
-    [dispatch, confirmAccount],
-  );
+  const onContinue = useCallback(() => {
+    if (selected === "moonpay") {
+      history.push("/platform/moonpay");
 
-  const confirmButtonTracking = useCallback(account => {
-    track("Buy Crypto Continue Button", {
-      currencyName: getAccountCurrency(account).name,
-      isEmpty: isAccountEmpty(account),
-    });
-  }, []);
+      return;
+    }
 
-  const allCurrencies = useCoinifyCurrencies("BUY");
-  const [provider] = useExchangeProvider();
+    setIsCoinify(true);
+  }, [selected, history]);
 
   return (
-    <BuyContainer>
+    <BuyContainer isCoinify={isCoinify}>
       <TrackPage category="Buy Crypto" />
-      {account ? (
-        <CoinifyWidget account={account} parentAccount={parentAccount} mode="buy" onReset={reset} />
+      {isCoinify ? (
+        <Coinify {...props} />
       ) : (
-        <SelectAccountAndCurrency
-          selectAccount={selectAccount}
-          allCurrencies={allCurrencies}
-          defaultCurrency={defaultCurrency}
-          defaultAccount={defaultAccount}
-          confirmCb={confirmButtonTracking}
-          provider={provider}
-          flow="buy"
-        />
+        <>
+          <Text ff="Inter|SemiBold" fontSize={18} color="palette.text.shade90">
+            {t("exchange.buy.title")}
+          </Text>
+          <Box horizontal width="100%" p="0 40.5px" my={48}>
+            <SelectProvider
+              provider="MoonPay"
+              cryptoCount={40}
+              onClick={() => toggle("moonpay")}
+              isActive={selected === "moonpay"}
+            >
+              <MoonPay size={48} />
+            </SelectProvider>
+            <SelectProvider
+              provider="Coinify"
+              cryptoCount={10}
+              onClick={() => toggle("coinify")}
+              isActive={selected === "coinify"}
+            >
+              <CoinifySquare size={48} />
+            </SelectProvider>
+          </Box>
+          <Footer>
+            <ContinueButton primary disabled={!selected} onClick={onContinue}>
+              <Trans i18nKey="common.continue" />
```

Comment:
Why using `Trans` where line 74 is utilizing `t()` ?

Reply:
Good catch! I was looking through the repo and found two different approaches so decided to test each. If I understand correctly <Trans> is still in the code since Component-based times and has to be refactored to use hook approach. 

Fixed, thanks.